web/client/plugins/__tests__/Share-test.jsx

```diff
@@ -230,14 +233,219 @@ describe('Share Plugin', () => {
+                expect(sharePanelDialog).toExist();
+                const switchButton = sharePanelDialog.querySelector('.mapstore-switch-btn input[type=\'checkbox\']');
+                expect(switchButton).toExist();
+                expect(switchButton.checked).toBe(true);
+                const checkBoxes = sharePanelDialog.querySelectorAll('.panel-body .checkbox input[type=\'checkbox\']');
+                const bboxCheckbox = checkBoxes[0];
+                const centerAndZoomCheckbox = checkBoxes[1];
+                expect(bboxCheckbox).toExist();
+                expect(centerAndZoomCheckbox).toExist();
+                expect(bboxCheckbox.checked).toBe(true);
+                expect(centerAndZoomCheckbox.checked).toBe(false);
+                done();
+            }, 500);
+
+            controls = {
+                share: {
+                    enabled: true,
+                    settings: {
+                        centerAndZoomEnabled: true
+                    }
+                }
+            };
+            const { Plugin: PluginModified } = getPluginForTest(SharePlugin, { controls, map });
+            ReactDOM.render(<PluginModified />, document.getElementById("container"));
+            setTimeout(() => {
+                const sharePanelDialog = document.getElementById('share-panel-dialog');
+                expect(sharePanelDialog).toExist();
+                const switchButton = sharePanelDialog.querySelector('.mapstore-switch-btn input[type=\'checkbox\']');
+                expect(switchButton).toExist();
+                expect(switchButton.checked).toBe(true);
+                const checkBoxes = sharePanelDialog.querySelectorAll('.panel-body .checkbox input[type=\'checkbox\']');
+                const bboxCheckbox = checkBoxes[0];
+                const centerAndZoomCheckbox = checkBoxes[1];
+                expect(bboxCheckbox).toExist();
+                expect(centerAndZoomCheckbox).toExist();
+                expect(bboxCheckbox.checked).toBe(false);
+                expect(centerAndZoomCheckbox.checked).toBe(true);
+                done();
+            }, 500);
+        } catch (e) {
+            done(e);
+        }
+    });
+
+    it('test Share plugin center and zoom params', (done) => {
+        const controls = {
+            share: {
+                enabled: true,
+                settings: {
+                    centerAndZoomEnabled: true
+                }
+            }
+        };
+        const map = {
+            center: {crs: "EPSG:4326",
+                x: -86.25,
+                y: 38.07
+            },
+            zoom: 5
+        };
+        const props = {
+            advancedSettings: {
+                centerAndZoom: true
+            }
+        };
+        const { Plugin } = getPluginForTest(SharePlugin, { controls, map });
+        ReactDOM.render(<Plugin {...props}/>, document.getElementById("container"));
+
+        setTimeout(() => {
```

Comment:
if possible do not use setTimeout in tests because they tend to be false positive,
for example try to to an expect(true).toBe(false) inside it, and check that this test should fail.

it it does not fail, then you need to fix these setTimeouts